rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -2,21 +2,32 @@
   'use strict';
   function RapidResponseAsideView(runtime, element) {
     var toggleStatusUrl = runtime.handlerUrl(element, 'toggle_block_open_status');
+    var responsesUrl = runtime.handlerUrl(element, 'responses');
     var $element = $(element);
 
     var rapidTopLevelSel = '.rapid-response-block';
     var rapidBlockContentSel = '#rapid-response-content';
     var toggleTemplate = _.template($(element).find("#rapid-response-toggle-tmpl").text());
 
-    function render(state) {
+    // default values
+    var state = {
+      is_open: false,
+      is_staff: false,
+      responses: []
+    };
+
+    function render() {
       // Render template
       var $rapidBlockContent = $element.find(rapidBlockContentSel);
       $rapidBlockContent.html(toggleTemplate(state));
 
       $rapidBlockContent.find('.problem-status-toggle').click(function(e) {
         $.get(toggleStatusUrl).then(
-          function(state) {
-            render(state);
+          function(newState) {
+            state = Object.assign({}, state, newState);
+            render();
+
+            pollForResponses();
```

Comment:
We only need to execute this if the problem has been newly opened, right? I know that if the problem is closed the polling function won't recurse, but this extra execution if the problem has just been closed seems unnecessary

Reply:
It would be necessary to get any responses submitted between the last poll and the problem being closed

Reply:
Unless I'm not reading this right, the polling logic is such that if the problem is still open after we hit the endpoint, we hit it again 3 seconds later, regardless of whether or not the problem is closed during those 3 seconds. Obviously it wouldn't be a huge performance hit to just leave this here, but it seems like we're covered for fetching responses after the problem is closed.

Reply:
Yes, you're right, I'll add the check